crd/multitenancy/api/v1alpha1/podnetwork.go

```diff
@@ -36,15 +37,31 @@ type PodNetworkList struct {
 	Items           []PodNetwork `json:"items"`
 }
 
+// +kubebuilder:validation:Enum=acn.azure.com/vnet-nic;acn.azure.com/gpu-nic
+type DeviceType string
+
+const (
+	DeviceTypeVnetNIC DeviceType = "acn.azure.com/vnet-nic"
+	DeviceTypeGPUNIC  DeviceType = "acn.azure.com/gpu-nic"
+)
+
 // PodNetworkSpec defines the desired state of PodNetwork
 type PodNetworkSpec struct {
+	// NetworkID is the identifier for the network, e.g. vnet guid or IB network ID
 	// +kubebuilder:validation:Optional
-	// customer vnet guid
-	VnetGUID string `json:"vnetGUID,omitempty"`
+	NetworkID string `json:"networkID,omitempty"`
+	// DeviceType is the device type that is required by this network
+	// +kubebuilder:validation:Optional
+	DeviceType DeviceType `json:"deviceType,omitempty"`
```

Comment:
neat: could we name this NetworkType? I see that the device determines the type of network it will live in, but why not just call it networkType?

Reply:
we want the values to match when we use device plugin, so orchestrator will pass, for example, `acn.azure.com/vnet-cni` to both `podNetwork.spec.deviceType` and the `pod.spec.resources.limits."acn.azure.com/vnet-cni"`. If we chose "networkType" instead, the orchestrators would then have to know about two separate magic values and maintain a mapping between them, which we don't want.